tests/NLog.UnitTests/LayoutRenderers/ExceptionDataLayoutRendererTests.cs

```diff
@@ -0,0 +1,316 @@
+                    <logger minlevel='Info' writeTo='debug1' />
+                </rules>
+            </nlog>");
+
+            var ex = new ExceptionWithBrokenMessagePropertyException();
+            var exRecorded = Record.Exception(() => logger.Error(ex, "msg"));
+            Assert.Null(exRecorded);
+        }
+
+        [Fact]
+        public void NoDatkeyTest()
+        {
+            LogManager.Configuration = XmlLoggingConfiguration.CreateFromXmlString(@"
+            <nlog>
+                <targets>
+                    <target name='debug1' type='Debug' layout='${exceptiondata:}' />
+                </targets>
+                <rules>
+                    <logger minlevel='Info' writeTo='debug1' />
+                </rules>
+            </nlog>");
+
+            var ex = GetNestedExceptionWithStackTrace("Goodbye World");
+            logger.Fatal(ex, "msg");
+            AssertDebugLastMessage("debug1", "");
+        }
+
+        [Fact]
+        public void NoDatkeyUingParamTest()
+        {
+            LogManager.Configuration = XmlLoggingConfiguration.CreateFromXmlString(@"
+            <nlog>
+                <targets>
+                    <target name='debug1' type='Debug' layout='${exceptiondata:DataKey=}' />
+                </targets>
+                <rules>
+                    <logger minlevel='Info' writeTo='debug1' />
+                </rules>
+            </nlog>");
+
+            var ex = GetNestedExceptionWithStackTrace("Goodbye World");
+            logger.Fatal(ex, "msg");
+            AssertDebugLastMessage("debug1", "");
+        }
+
+        [Fact]
+        public void BaseExceptionTest()
+        {
+            const string exceptionDataKey = "testkey";
+            const string exceptionDataValue = "testvalue";
+            LogManager.Configuration = XmlLoggingConfiguration.CreateFromXmlString(@"
+            <nlog>
+                <targets>
+                    <target name='debug1' type='Debug' layout='${exceptiondata:DataKey=" + exceptionDataKey + @", BaseException=false}' />
+                </targets>
+                <rules>
+                    <logger minlevel='Info' writeTo='debug1' />
+                </rules>
+            </nlog>");
+
+            var ex = GetNestedExceptionWithStackTrace("Goodbye World");
+            ex.Data.Add(exceptionDataKey, exceptionDataValue);
+            logger.Fatal(ex, "msg");
+            AssertDebugLastMessage("debug1", exceptionDataValue);
+        }
+
+        private class ExceptionWithBrokenMessagePropertyException : NLogConfigurationException
+        {
+            public override string Message => throw new Exception("Exception from Message property");
+        }
+
+        /// <summary>
+        /// Get an exception with stacktrace by generating a exception
+        /// </summary>
+        /// <param name="exceptionMessage"></param>
+        /// <returns></returns>
+        private Exception GetExceptionWithStackTrace(string exceptionMessage)
```

Comment:
This funky method used to generate exceptions with exotic stacktraces doesn't seem relevant for doing Data-collection-lookups. I can understand the need for nested exceptions to test BaseException-property, but the exception doesn't need to be thrown from a special context to test this. Better with simple arrange + act + assert.